packages/wdio-cucumber-framework/src/constants.ts

```diff
@@ -1,50 +1,40 @@
-import type { Pickle, PickleStep } from '@cucumber/messages'
+import url from 'node:url'
+import path from 'node:path'
+
 import type { CucumberOptions } from './types.js'
 
 export const DEFAULT_TIMEOUT = 60000
 
-export const DEFAULT_OPTS: CucumberOptions = {
-    backtrace: false, // <boolean> show full backtrace for errors
-    requireModule: [], // <string[]> ("module") require MODULE files (repeatable)
-    failAmbiguousDefinitions: false, // <boolean> treat ambiguous definitions as errors
-    failFast: false, // <boolean> abort the run on first failure
-    ignoreUndefinedDefinitions: false, // <boolean> treat undefined definitions as warnings
-    names: [], // <(string|RegExp)[]> only execute the scenarios with name matching the expression (repeatable)
-    require: [], // <string> (file/dir/glob) require files before executing features
-    order: 'defined', // <string> switch between deterministic  and random feature execution. Either "defined", "random" or "random:42" whereas 42 is the seed for randomization
-    snippets: true, // <boolean> hide step definition snippets for pending steps
-    source: true, // <boolean> hide source uris
-    strict: false, // <boolean> fail if there are any undefined or pending steps
-    tagExpression: '', // <string> (expression) only execute the features or scenarios with tags matching the expression
-    tagsInTitle: false, // <boolean> add cucumber tags to feature or scenario name
-    timeout: DEFAULT_TIMEOUT, // <number> timeout for step definitions in milliseconds
-    retry: 0,
-    scenarioLevelReporter: false,
-    featureDefaultLanguage: 'en'
-}
-
-/* istanbul ignore next */
-export const NOOP = function () {}
+export const FILE_PROTOCOL = 'file://'
 
-export const CUCUMBER_HOOK_DEFINITION_TYPES = [
-    'beforeTestRunHookDefinitionConfigs',
-    'beforeTestCaseHookDefinitionConfigs',
-    'afterTestCaseHookDefinitionConfigs',
-    'afterTestRunHookDefinitionConfigs',
-] as const
+const cucumberFormatter = path.resolve(url.fileURLToPath(import.meta.url), '../cucumberFormatter.js')
```

Comment:
Can we just use a file url string for windows and linux based OS?
```suggestion
const cucumberFormatter = url.pathToFileURL(path.resolve(url.fileURLToPath(import.meta.url), '../cucumberFormatter.js')).href
```

Reply:
At first, I took the same approach. When we use `url.pathToFileURL`, the file URL comes with the `file:///` prefix with three slashes (`///`), which works well for Linux-based systems. However, Cucumber internally modifies the provided file URL and raises an error if there are three slashes (`///`) present on windows. 

Upon further investigation, I observed that Cucumber employs the split [operation](https://github.com/cucumber/cucumber-js/blob/85b1e5ed13d73d037d39641399f3fdff66faa023/src/configuration/option_splitter.ts#L5), which creates issues on Windows. As a result, I've explicitly added a double slash prefix to the file path for Windows.

Reply:
Sorry this is such a pain point with Cucumber!

I just released 9.5.1 of cucumber-js which includes support for expressing each format as a 1- or 2-item array instead of a string, so it's explicit what is the formatter name/path and (optionally) what is the target and there is no split operation. So with the configuration below it could be something like:

```json
format: [
  [cucumberFormatter]
]
```

And that value should be fine as a regular file URL regardless of platform - if you're able to give this a try, let me know how it goes.

Also more generally, feel free to start a thread in https://github.com/cucumber/cucumber-js/discussions/2041 with any feedback you have on the new API. I'm conscious that the docs are pretty thin still, so I'm happy to fill any gaps.

Reply:
Thanks @davidjgoss for handling this. I'll make the necessary changes and will get in touch if I encounter any issues.